core/internal/runbranch/rewind.go

```diff
@@ -0,0 +1,168 @@
+package runbranch
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/wandb/simplejsonext"
+	"github.com/wandb/wandb/core/internal/filestream"
+	"github.com/wandb/wandb/core/internal/gql"
+	"github.com/wandb/wandb/core/pkg/service"
+	"github.com/wandb/wandb/core/pkg/utils"
+)
+
+type RewindBranch struct {
+	ctx    context.Context
+	client graphql.Client
+	branch BranchPoint
+}
+
+func NewRewindBranch(
+	ctx context.Context,
+	client graphql.Client,
+	runid string,
+	metricName string,
+	metricValue float64,
+) *RewindBranch {
+	return &RewindBranch{
+		ctx:    ctx,
+		client: client,
+		branch: BranchPoint{
+			RunID:       runid,
+			MetricName:  metricName,
+			MetricValue: metricValue,
+		},
+	}
+}
+
+//gocyclo:ignore
```

Comment:
this function should be refactored instead of disabling complexity checks

Reply:
ah i don't remember adding this